### PR TITLE
DIT 805 - Pagination

### PR DESCRIPTION
**Pull request by ed0906**

### CODE REVIEW

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/cases_list.scala.html [hunk 1]
```diff
@@ -57,6 +57,6 @@
     </tbody>
 </table>
 
-@if(!cases.exists(_.application.isBTI)) {
+@if(cases.isEmpty) {
```
Comment (ed0906) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/cases_list.scala.html]: This is why we need the extra `application_type` filter on the back end

File: app/assets/stylesheets/layout.scss [hunk 2]
```diff
@@ -37,6 +37,15 @@
   margin-top: 30px;
 }
 
+.mt-45 {
```
Comment (pllaurado) [inline app/assets/stylesheets/layout.scss]: not a big fan of em? ;-)

File: app/assets/stylesheets/search.scss [hunk 3]
```diff
@@ -26,4 +26,8 @@
     top: 18px;
   }
 
-}
\ No newline at end of file
+}
+
+//table {
```
Comment (pllaurado) [inline app/assets/stylesheets/search.scss]: Maybe remove it?

File: app/assets/stylesheets/layout.scss (hunk 2 above)
Comment (ed0906) [reply app/assets/stylesheets/layout.scss]: Just being consistent with what's already there :D

File: app/assets/stylesheets/search.scss (hunk 3 above)
Comment (ed0906) [reply app/assets/stylesheets/search.scss]: I suspect in post amigos Joe will notice the table is wider than it should be, which is when I can uncomment this

File: app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala [hunk 4]
```diff
@@ -44,17 +44,17 @@ class BindingTariffClassificationConnector @Inject()(appConfig: AppConfig, clien
     client.GET[Option[Case]](url)
   }
 
-  def findCasesByQueue(queue: Queue)(implicit hc: HeaderCarrier): Future[Seq[Case]] = {
+  def findCasesByQueue(queue: Queue, pagination: Pagination)(implicit hc: HeaderCarrier): Future[Paged[Case]] = {
     val queueId = if (queue == Queues.gateway) "none" else queue.id
-    val queryString = s"queue_id=$queueId&assignee_id=none&status=$statuses&sort_by=days-elapsed&sort_direction=desc"
+    val queryString = s"application_type=BTI&queue_id=$queueId&assignee_id=none&status=$statuses&sort_by=days-elapsed&sort_direction=desc&page=${pagination.page}&page_size=${pagination.pageSize}"
```
Comment (pllaurado) [inline app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala]: I can't find any QueryBuilder in our libraries but we maybe need to keep in mind that this string is not to friendly

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/PaginationUtil.scala [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.views.partials
+
+import scala.collection.immutable.Seq
+
+object PaginationUtil {
```
[the remark this answers is not in the document]
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/PaginationUtil.scala]: It's to generate the `Previous 1 2 3 4 5 6 Next`

[the remark this answers is not in the document]
Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/PaginationUtil.scala]: I tried various strategies, ended up going with this... :(

File: app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/PaginationUtil.scala [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  def surroundingPages(pageIndex: Int, pageCount: Int, available: Int): Seq[Int] = {
+    var indexes: Seq[Int] = Seq(pageIndex)
+    var surrounded = surround(indexes, pageCount)
```
Comment (gokyo) [inline app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/PaginationUtil.scala]: `var` ?

**Pull request merged**